pandas/io/formats/style.py

```diff
@@ -1222,6 +1222,78 @@ class MyStyler(cls):
 
         return MyStyler
 
+    def pipe(self, func, *args, **kwargs):
+        """
+        Apply ``func(self, *args, **kwargs)``, and return the result.
+
+        .. versionadded:: 0.24.0
+
+        Parameters
+        ----------
+        func : function
+            Function to apply to the Styler.
+            ``*args``, and ``**kwargs`` are passed into ``func``.
+            Alternatively a ``(callable, data_keyword)`` tuple where
+            ``data_keyword`` is a string indicating the keyword of
+            ``callable`` that expects the Styler.
+        *args : iterable, optional
+            Positional arguments passed into ``func``.
+        **kwargs : dict, optional
+            Dictionary of keyword arguments passed into ``func``.
+
+        Returns
+        -------
+        result : object
+            The value returned by ``func``.
+
+        See Also
+        --------
+        DataFrame.pipe : Analogous method for DataFrame.
+        Styler.apply : Apply a function row-wise, column-wise, or table-wise to
+            modify the dataframe's styling.
+
+        Notes
+        -----
+        Like :meth:`DataFrame.pipe`, this method can simplify the
+        application of several user-defined functions to a styler.  Instead
+        of writing:
+
+        .. code-block:: python
+
+            f(g(df.style.set_precision(3), arg1=a), arg2=b, arg3=c)
+
+        users can write:
+
+        .. code-block:: python
+
+            (df.style.set_precision(3)
+               .pipe(g, arg1=a)
+               .pipe(f, arg2=b, arg3=c))
+
+        In particular, this allows users to define functions that take a
+        styler object, along with other parameters, and return the styler after
+        making styling changes (such as calling :meth:`Styler.apply` or
+        :meth:`Styler.set_properties`).  Using ``.pipe``, these user-defined
+        style "transformations" can be interleaved with calls to the built-in
+        Styler interface.
+
+        Examples
+        --------
+        >>> def set_standard_formatting(styler):
+        ...     return (styler.set_properties(**{'text-align': 'right'})
+        ...                   .format({'X': '{:.1%}'}))
+
+        The user-defined highlight function above can be called within a
+        sequence of other style modifications:
+
+        >>> df = pd.DataFrame({'A': list(range(-1, 4)), 'X': np.arange(0.2, 1.2, 0.2)})
```

Comment:
Use something that looks real, no `range` fuctions, and no `a`, `x`, `foo` columns. We start to have many examples with animal names (e.g. `cat`, `penguin`, `falcon`...). I'd use that for consistency (hopefully at some point we have all the examples with similar data). Just have couple of columns that are percentages, and 2 or 3 rows.

Then rename `set_standard_formatting` to `percentage_format` (if I understand correctly, that's what it is, right?

Reply:
I made some changes here, and fixed the reference in the text, which was incorrect.  I changed the function name as you suggested, while trying to make it clear that the function was user-defined and application-specific.

I'll be honest, using animal names in examples seems a little silly to me, but that's up to the pandas developers.  

In this case, using numeric data rather than strings made more sense to me, since numeric data presents styling/formatting choices, and special values (min/max/null) can be highlighted.